core/src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -1122,13 +1126,19 @@ public void clearRequestCache(IndexShard shard) {
      */
     public void loadIntoContext(ShardSearchRequest request, SearchContext context, QueryPhase queryPhase) throws Exception {
         assert canCache(request, context);
-        final IndexShardCacheEntity entity = new IndexShardCacheEntity(context.indexShard(), out -> {
+        final DirectoryReader directoryReader = context.searcher().getDirectoryReader();
+        
+        Tuple<Boolean, BytesReference> cacheResult = cacheShardLevelResult(context.indexShard(), directoryReader, request.cacheKey(), out -> {
             queryPhase.execute(context);
-            context.queryResult().writeToNoId(out);
+            try {
```

Comment:
Can we just use a boolean here that we set to false when this  loader is executed?